src/Utility/Text.php

```diff
@@ -1054,38 +1063,86 @@ public static function parseFileSize($size, $default = false)
     /**
      * Get default transliterator identifier string.
      *
+     * @deprecated 3.7.0 Use getTransliterator() instead.
      * @return string Transliterator identifier.
      */
     public static function getTransliteratorId()
     {
-        return static::$_defaultTransliteratorId;
+        deprecationWarning(
+            'Text::getTransliteratorId() is deprecated. ' .
+            'Use Text::getTransliterator() instead.'
+        );
```

Comment:
This is a fairly new method.  Deprecating it now seems a bit indecisive and creates unnecessary API churn. Why can't we keep both the ID and transliterator instance? The `transliterator_transliterate` will work with either one, and internally we can prefer the object over the id.

Reply:
There isn't really a technical reason not to do that, I just thought that such internal weighting of properties might not be overly welcome, as there are quite a few other methods that seem to prefer union types, but if separate methods are preferred in this case, then I'll gladly change it.

Reply:
Would you prefer separate options for `Text::slug()` too?

Reply:
I think separate options would be better. We could raise an error if users send both options.

Reply:
I gave it a try, but somehow having two options feels weird... they would both be able to trigger the default behavior when `null` is passed, while the possibly set transliterator object would get preferred by `Text::transliterate()`, also they would both be able to disable transliteration when `false` is passed.

Not sure what users would expect if they for example pass `transliteratorId => null`, would it be reasonable if they'd assume that the default transliterator _identifier_ is used, instead of the possibly set default transliterator _object_?

Reply:
> Deprecating it now seems a bit indecisive and creates unnecessary API churn

If the new approach here is the better one, we shouldnt be afraid of fixing the API once more - hopefully the last time. So deprecation + having a single one forward seems to be reasonable none the less.

Reply:
The current transliteratorId support was added in 3.2, which wasn't that long ago. Isn't the transliteratorId something that can be converted into an instance when the id is set? Same with the get, the id can be derived from the object. Does only accepting one or the other option for `transliterate()` make the API simpler?

I'm pushing on this because I think we're sometimes eager to deprecate and replace which can often become change without improvement for end users.

Reply:
So you are saying this is already feature complete and this change here isn't necessary at all anymore given the current API?

Reply:
Just to make it clear, I was referring to the options for `Text::slug()` in my last comment! I'm less concerned about separate methods, even though I think a single getter/setter should be the goal at some point.

As far as the options go, I think it would be more straight forward with a single option that accepts both, identifiers and objects, just like `Text::transliterate()` can accept both. If deprecation is a concern there too, then the `transliteraterId` option could also be kept as an alias for the time being?

> Isn't the transliteratorId something that can be converted into an instance when the id is set? Same with the get, the id can be derived from the object.
	
Transliterator objects can be created from identifiers, and identifiers can be retrieved from the objects, but objects built from rules don't expose valid identifiers (they can't, as identifiers must refer to ICU transliterators), they'd always return `RulesTransPHP` as the identifier.

Reply:
> Transliterator objects can be created from identifiers, and identifiers can be retrieved from the objects, but objects built from rules don't expose valid identifiers ... they'd always return RulesTransPHP as the identifier.

I think that's a reasonable trade-off to make.

> I think it would be more straight forward with a single option that accepts both, identifiers and objects

I'm on-board with that approach too.

Reply:
Ok. Just so I don't misunderstand it, because I'm a little confused, it should be separate getters/setters for identifier and object (where objects are derived from identifiers, and identifiers are derived from objects in case possible), and a single option (with or without deprecation notice?) for `Text::slug()`?

Reply:
I think the use case would be the same as the one for accepting objects in the first place, the need to apply custom rules. Wouldn't it be a little weird if only identifiers could be applied selectively? I'll push an update so we have something to look at.